instagram/src/components/CommentSection/CommentSection.js

```diff
@@ -1,14 +1,33 @@
-import React from 'react';
+import React, {Component} from 'react';
 import Comment from './Comment'
 
-const CommentSection = props => {
-    console.log(props.comments);
-    return <div>{props.comments.map((c, i) =>
-         <Comment username={c.username} key={i} text={c.text} />
-    )}</div>;
+class CommentSection extends Component {
+    constructor(props){
+        super(props);
+        this.state = {
+            comments: [],
+        }
+    }
 
+    componentDidMount(){
+        this.setState({comments: this.props.comments})
```

Comment:
since setState calls sometimes happen asynchronously with each other, there's actually a cleaner way to do this: because this piece of data is already coming in as props, which is passed through your constructor, you can immediately set it on state within the constructor without an external CDM hook like here. so `this.state = {comments: props.comments}`